File: content/examples/other/leetcode/problems/insert-interval/example-2/solution.js
function insertInterval(intervals, newInterval) {
  // Same optimal linear scan; written with minimal temporary state.
  const out = [];
  let [ns, ne] = newInterval;
  for (const [s, e] of intervals) {
    if (e < ns) out.push([s, e]);
    else if (s > ne) {
      out.push([ns, ne]);
      ns = s;
      ne = e;
    } else {
      ns = Math.min(ns, s);
      ne = Math.max(ne, e);
    }
  }
  out.push([ns, ne]);
  return out;
}

if (require.main === module) {
  console.log(insertInterval([[1, 3], [6, 9]], [2, 5]));
}

module.exports = { insertInterval };

File: content/examples/other/leetcode/problems/insert-interval/example-2/solution.py
def insert_interval(intervals, new_interval):
    out = []
    ns, ne = new_interval
    for s, e in intervals:
        if e < ns:
            out.append([s, e])
        elif s > ne:
            out.append([ns, ne])
            ns, ne = s, e
        else:
            ns = min(ns, s)
            ne = max(ne, e)
    out.append([ns, ne])
    return out


if __name__ == "__main__":
    print(insert_interval([[1, 3], [6, 9]], [2, 5]))
